debugger/src/debug_adapter/protocol.rs

```diff
@@ -344,7 +363,16 @@ impl<R: Read, W: Write> ProtocolAdapter for DapAdapter<R, W> {
 
         let encoded_resp = serde_json::to_vec(&resp)?;
 
-        self.send_data(&encoded_resp)?;
+        match self.send_data(&encoded_resp) {
+            Ok(_) => {}
+            Err(error) => {
+                // Store pending request so that we can notify the listener about the error.
+                self.output.flush()?;
+                self.pending_requests
+                    .insert(request.seq, request.command.clone());
+                return self.send_response::<()>(request, Err(DebuggerError::StdIO(error)));
```

Comment:
I'm struggling to see where this would work. If there was an error sending data, why should sending an error message over the same channel work? I would assume that sending the error message would fail as well, leading to an potential endless loop.

Reply:
Actually, if there was an error because of "would block", then the next one won't necessarily fail, but I see your point. What do you suggest for this case?

Reply:
We should use a different function at least, to avoid a potential endless recursion. I would honestly just stop, since there is not much we can do if the communication channel breaks.

Reply:
Fair enough ... I will log and propagate the error, which will terminate the debug session.

Reply:
Done.